File: transaction-analyzer/kotlin/src/test/kotlin/Fixtures.kt
import java.math.BigDecimal
import java.time.LocalDateTime
import java.time.format.DateTimeFormatter

object Fixtures {
    val transactionData = listOf(
        Transaction(
            id = "WLMFRDGD",
            transactedAt = LocalDateTime.parse(
                "20/08/2020 12:45:33",
                DateTimeFormatter.ofPattern("dd/MM/yyyy HH:mm:ss")
            ),
            amount = BigDecimal.valueOf(59.99),
            merchantName = "Kwik-E-Mart",
            type = TransactionType.PAYMENT,
            relatedTransactionId = ""
        ),
        Transaction(
            id = "YGXKOEIA",
            transactedAt = LocalDateTime.parse(
                "20/08/2020 12:46:17",
                DateTimeFormatter.ofPattern("dd/MM/yyyy HH:mm:ss")
            ),
            amount = BigDecimal.valueOf(10.95),
            merchantName = "Kwik-E-Mart",
            type = TransactionType.PAYMENT,
            relatedTransactionId = ""
        ),
        Transaction(
            id = "LFVCTEYM",
            transactedAt = LocalDateTime.parse(
                "20/08/2020 12:50:02",
                DateTimeFormatter.ofPattern("dd/MM/yyyy HH:mm:ss")
            ),
            amount = BigDecimal.valueOf(5.00),
            merchantName = "MacLaren",
            type = TransactionType.PAYMENT,
            relatedTransactionId = ""
        ),
        Transaction(
            id = "SUOVOISP",
            transactedAt = LocalDateTime.parse(
                "20/08/2020 13:12:22",
                DateTimeFormatter.ofPattern("dd/MM/yyyy HH:mm:ss")
            ),
            amount = BigDecimal.valueOf(5.00),
            merchantName = "Kwik-E-Mart",
            type = TransactionType.PAYMENT,
            relatedTransactionId = ""
        ),
        Transaction(
            id = "AKNBVHMN",
            transactedAt = LocalDateTime.parse(
                "20/08/2020 13:14:11",
                DateTimeFormatter.ofPattern("dd/MM/yyyy HH:mm:ss")
            ),
            amount = BigDecimal.valueOf(10.95),
            merchantName = "Kwik-E-Mart",
            type = TransactionType.REVERSAL,
            relatedTransactionId = "YGXKOEIA"
        ),
        Transaction(
            id = "JYAPKZFZ",
            transactedAt = LocalDateTime.parse(
                "20/08/2020 14:07:10",
                DateTimeFormatter.ofPattern("dd/MM/yyyy HH:mm:ss")
            ),
            amount = BigDecimal.valueOf(99.50),
            merchantName = "MacLaren",
            type = TransactionType.PAYMENT,
            relatedTransactionId = ""
        )
    )

}

File: transaction-analyzer/php/tests/Fixtures.php
<?php

namespace TransactionAnalyser\Tests;

use Brick\DateTime\LocalDateTime;
use Brick\Math\BigDecimal;
use DateTime;
use TransactionAnalyser\Transaction;
use TransactionAnalyser\TransactionType;

class Fixtures
{
    static function transactionData(): array
    {
        return [
            new Transaction(
                "WLMFRDGD",
                LocalDateTime::fromNativeDateTime(DateTime::createFromFormat('d/m/Y H:i:s', "20/08/2020 12:45:33")),
                BigDecimal::of(59.99),
                "Kwik-E-Mart",
                TransactionType::PAYMENT,
                ""
            ),
            new Transaction(
                "YGXKOEIA",
                LocalDateTime::fromNativeDateTime(DateTime::createFromFormat('d/m/Y H:i:s', "20/08/2020 12:46:17")),
                BigDecimal::of(10.95),
                "Kwik-E-Mart",
                TransactionType::PAYMENT,
                ""
            ),
            new Transaction(
                "LFVCTEYM",
                LocalDateTime::fromNativeDateTime(DateTime::createFromFormat('d/m/Y H:i:s', "20/08/2020 12:50:02")),
                BigDecimal::of(5.00),
                "MacLaren",
                TransactionType::PAYMENT,
                ""
            ),
            new Transaction(
                "SUOVOISP",
                LocalDateTime::fromNativeDateTime(DateTime::createFromFormat('d/m/Y H:i:s', "20/08/2020 13:12:22")),
                BigDecimal::of(5.00),
                "Kwik-E-Mart",
                TransactionType::PAYMENT,
                ""
            ),
            new Transaction(
                "AKNBVHMN",
                LocalDateTime::fromNativeDateTime(DateTime::createFromFormat('d/m/Y H:i:s', "20/08/2020 13:14:11")),
                BigDecimal::of(10.95),
                "Kwik-E-Mart",
                TransactionType::REVERSAL,
                "YGXKOEIA"
            ),
            new Transaction(
                "JYAPKZFZ",
                LocalDateTime::fromNativeDateTime(DateTime::createFromFormat('d/m/Y H:i:s', "20/08/2020 14:07:10")),
                BigDecimal::of(99.50),
                "MacLaren",
                TransactionType::PAYMENT,
                ""
            ),
        ];
    }
}